text/0118-search-api.md

```diff
@@ -108,156 +109,344 @@ It returns the number of documents matching the current search query for each gi
+A custom crop length set in this way has priority over the `cropLength` parameter.
 
-> See [_Formatted Field Behavior](0039-_formatted-field-behavior_.md)
+##### 3.1.11.2. searchableAttributes
 
-##### 1.2.1.10 `cropLength`
+As for `attributesToHighlight`, attributes not defined in the `searchableAttributes` index setting are also cropped if assigned to `attributesToCrop`.
+
+##### 3.1.11.3. stopWords
+
+Attributes defined in the `stopWords` index setting are counted as words regarding `cropLength`.
+
+##### 3.1.11.3. Tokenizer Separators
+
+Tokenizer separators aren't counted as words regarding `cropLength`.
+
+#### 3.1.12. `cropLength`
 
 - Type: Integer
 - Required: False
-- Default: `200`
+- Default: `10`
+
+Sets the total number of **words** to keep for the cropped part of an attribute specified in the `attributesToCrop` parameter. It means that if `10` is set for `cropLength`, the cropped part returned in `_formatted` will only be 10 words long.
+
+This parameter is taken into account when `attributesToCrop` is specified. See [3.1.11. `attributesToCrop`](#3111-attributestocrop) section.
+
+- 🔴 Sending a value with a different type than `Integer` or `null` for `cropLength` returns a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+
+##### 3.1.12.1 Examples
+
+###### 3.1.12.1.1. Extending around
+
+Given an attribute defined in `attributesToCrop` containing:
+
+`"In his ravenous hatred he found no peace, and with boiling blood he scoured the umbral plains, seeking vengence afgainst the dark lords who had robbed him."`
+
+With `croplength` defined as `5` and `q` defined as `boiling blood`, the cropped value will be:
+
+`"…and with boiling blood he…"`
+
+Croppped query terms are counted as a word regarding `cropLength`.
+
+###### 3.1.12.1.3. Length Reassignment to the left.
 
-Configures the number of characters to keep on each side of the matching query term when using the `attributesToCrop` parameter.
+Given an attribute defined in `attributesToCrop` containing:
+
+`"In his ravenous hatred he found no peace, and with boiling blood he scoured the umbral plains, seeking vengence afgainst the dark lords who had robbed him."`
+
+With `croplength` defined as `5` and `q` defined as `had robbed`, the cropped value will be:
+
+`"…lords who had robbed him."`
+
+In this example:
+
+- `"had robbed"` count for `2` words.
+- The right part contains `1` words. e.g. `"him.`.
+- The cropped part will be filled by `2` words on the left to reach `cropLength`. e.g. `"lords who"`.
+
+###### 3.1.12.1.2. Length Reassignment to the right.
+
+Given an attribute defined in `attributesToCrop` containing:
+
+`"In his ravenous hatred he found no peace, and with boiling blood he scoured the umbral plains, seeking vengence afgainst the dark lords who had robbed him."`
+
+With `croplength` defined as `5` and `q` defined as `hatred`, the cropped value will be:
+
+`"In his ravenous hatred he…"`
```

Comment:
> "In his ravenous hatred he found no peace, and with boiling blood he scoured the umbral plains, seeking vengence afgainst the dark lords who had robbed him."

if the word is `he` I feel like with your examples the words around it are more often in front of the word than behind.

For example: 
```
 With boiling blood he scoured...
```

Instead of for example

```
blood he scoured the umbrella...
```

its a fake example just. In the first one 3 words are in front of `he` and in the second 3 word are behind `he`. 
What is the rule for this?

Reply:
@ManyTheFish Regarding https://github.com/meilisearch/specifications/pull/120#discussion_r837419461